### ISSUE TITLE
Is it possible to disable automatic detection of simple pure methods?

### ISSUE THREAD

**Issue by janopae** [vimeo/psalm#9665]
This is possible currently: https://psalm.dev/r/b2a9011f65

However, the library might change the following way, making it break: https://psalm.dev/r/7e645e6b69

This could be solved by making Psalm require to always explicitly define pureness as part of the method signature. At least, it would be sane to have a warning on missing pure annotations.

**Comment by weirdan** [comment]
I don't think we have that yet. If you want to try your hand at it, it happens here:

https://github.com/vimeo/psalm/blob/e2957a163190c1f5bd057f4ecbaab4b96098a94a/src/Psalm/Internal/PhpVisitor/Reflector/FunctionLikeNodeScanner.php#L247-L270

**Comment by ygottschalk** [comment]
Can `psalter` add those annotations? Would be nice...

**Comment by orklah** [comment]
Seems undocumented but yes it can. `--issues=MissingPureAnnotation` will do the trick
